### PR TITLE
[APPENG-1360] use lintly

### PR DESCRIPTION
**Pull request by jamesarosen**
* add editorconfig
 * remove jshint
 * add @fastly/prettier-config and dependencies
 * add @fastly/estlint-config and dependencies
 * add prettier and eslint config files
 * change `package.json` to use eslint for linting
 * de-lint

### CODE REVIEW

File: .eslintrc.js [hunk 1]
```diff
@@ -0,0 +1,9 @@
+module.exports = {
+  env: { node: true },
+  extends: ['@fastly'],
+
+  // handled by prettier:
+  rules: {
+    'generator-star-spacing': 0,
```
Comment (jamesarosen) [inline .eslintrc.js]: @psbanka we need to get this into lintly

File: .editorconfig [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+[*]
+end_of_line = lf
+charset = utf-8
+trim_trailing_whitespace = true
+insert_final_newline = true
+indent_style = space
+indent_size = 2
+
+[*.{diff,md}]
+trim_trailing_whitespace = false
```
Comment (psbanka) [inline .editorconfig]: Odd. Why wouldn't we want this?

File: commands/verify.js [hunk 3]
```diff
@@ -1,99 +1,117 @@
...
-      cli.error(body);
-    }
-  });
+        cli
+          .prompt('Type the approval domain to use (or ENTER if only 1): ')
+          .then(function(approval) {
+            let valid = json.metadata.valid_approvals.indexOf(approval)
 
-}
+            if (valid == -1) {
```
Comment (psbanka) [inline commands/verify.js]: doesn't double-equal throw a lint error?

File: .editorconfig (hunk 2 above)
Comment (jamesarosen) [reply .editorconfig]: I guess it came from [here](https://github.com/ember-cli/ember-cli/blob/master/blueprints/app/files/.editorconfig#L19-L20). I don't care either way on `.md`. `.diff` files should probably remain intact.

File: commands/verify.js (hunk 3 above)
Comment (jamesarosen) [reply commands/verify.js]: You'd hope so.

[@fastly/eslint-config](https://github.com/fastly/lintly/blob/main/eslint-config/index.js) doesn't define `eqeqeq` either way.

eslint-config-esnext [enables `eqeqeq`](https://github.com/kunalgolani/eslint-config/blob/41b2ad0b797b5ea4fdbd4d7ad8b1d033e660f191/packages/esnext/index.yaml#L26).

I'm not sure why this isn't caught.

File: .eslintrc.js (hunk 1 above)
Comment (jamesarosen) [reply .eslintrc.js]: Done in fastly/lintly#17

**Pull request merged**